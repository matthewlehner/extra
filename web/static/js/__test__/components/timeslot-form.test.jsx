import React from "react";
import renderer from "react-test-renderer";

import TimeslotForm from "components/timeslot-form";

const recurrenceValues = [{ name: "MONDAY" }, { name: "EVERYDAY" }];

describe("TimeslotForm component", () => {
  it("renders a form", () => {
    const props = {
      recurrences: recurrenceValues,
      collections: [{ id: "1", name: "Cool bud" }, { id: "25", name: "Misc." }],
      onCancel: jest.fn(),
      onSubmit: jest.fn()
    };

    const tree = renderer.create(<TimeslotForm {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
