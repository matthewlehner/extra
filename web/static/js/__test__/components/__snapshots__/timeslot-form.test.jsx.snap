// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeslotForm component renders a form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="time"
    onChange={[Function]}
    placeholder="00:00"
    required={true}
    type="text"
    value="01:30"
  />
  <div
    className="select"
  >
    <select
      className="form__control"
      id="recurrence"
      name="recurrence"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option>
        —
      </option>
      <option
        value="MONDAY"
      >
        Monday
      </option>
      <option
        value="EVERYDAY"
      >
        Everyday
      </option>
    </select>
  </div>
  <div
    className="select"
  >
    <select
      className="form__control"
      id="collectionId"
      name="collectionId"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option>
        —
      </option>
      <option
        value="1"
      >
        Cool bud
      </option>
      <option
        value="25"
      >
        Misc.
      </option>
    </select>
  </div>
  <button
    className="linkButton"
    onClick={[Function]}
  >
    Add Timeslot
  </button>
  <button
    className="linkButton"
    onClick={[Function]}
  >
    Cancel
  </button>
</form>
`;
